node/external/blockAPI/metaBlock_test.go

```diff
@@ -596,3 +604,132 @@ func TestMetaAPIBlockProcessor_GetBlockByRound_GetBlockByNonce_EpochStartBlock(t
+		t.Parallel()
+
+		headerHash := []byte("d08089f2ab739520598fd7aeed08c427460fe94f286383047f3f61951afc4e00")
+
+		storerMock := genericMocks.NewStorerMockWithEpoch(1)
+		metaAPIBlockProc := createMockMetaAPIProcessor(
+			headerHash,
+			storerMock,
+			true,
+			true,
+		)
+
+		res, err := metaAPIBlockProc.GetAlteredAccountsForBlock(api.GetAlteredAccountsForBlockOptions{})
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "not found")
+		require.Nil(t, res)
+	})
+
+	t.Run("get altered account by block hash - should work", func(t *testing.T) {
+		t.Parallel()
+
+		marshaller := &testscommon.MarshalizerMock{}
+		headerHash := []byte("d08089f2ab739520598fd7aeed08c427460fe94f286383047f3f61951afc4e00")
+		mbHash := []byte("mb-hash")
+		txHash0, txHash1 := []byte("tx-hash-0"), []byte("tx-hash-1")
+
+		mbhReserved := block.MiniBlockHeaderReserved{}
+
+		mbhReserved.IndexOfLastTxProcessed = 1
+		reserved, _ := mbhReserved.Marshal()
+
+		metaBlock := &block.MetaBlock{
+			Nonce: 37,
+			Epoch: 1,
+			MiniBlockHeaders: []block.MiniBlockHeader{
+				{
+					Hash:     mbHash,
+					Reserved: reserved,
+				},
+			},
+		}
+		miniBlock := &block.MiniBlock{
+			TxHashes: [][]byte{txHash0, txHash1},
+		}
+		tx0 := &transaction.Transaction{
+			SndAddr: []byte("addr0"),
+			RcvAddr: []byte("addr1"),
+		}
+		tx1 := &transaction.Transaction{
+			SndAddr: []byte("addr2"),
+			RcvAddr: []byte("addr3"),
+		}
+		miniBlockBytes, _ := marshaller.Marshal(miniBlock)
+		metaBlockBytes, _ := marshaller.Marshal(metaBlock)
+		tx0Bytes, _ := marshaller.Marshal(tx0)
+		tx1Bytes, _ := marshaller.Marshal(tx1)
+
+		storerMock := genericMocks.NewStorerMockWithEpoch(1)
+		_ = storerMock.Put(headerHash, metaBlockBytes)
+		_ = storerMock.Put(mbHash, miniBlockBytes)
+		_ = storerMock.Put(txHash0, tx0Bytes)
+		_ = storerMock.Put(txHash1, tx1Bytes)
+
+		metaAPIBlockProc := createMockMetaAPIProcessor(
+			headerHash,
+			storerMock,
+			true,
+			true,
+		)
+
+		metaAPIBlockProc.apiTransactionHandler = &mock.TransactionAPIHandlerStub{
+			UnmarshalTransactionCalled: func(txBytes []byte, _ transaction.TxType) (*transaction.ApiTransactionResult, error) {
+				var tx transaction.Transaction
+				_ = marshaller.Unmarshal(&tx, txBytes)
+
+				return &transaction.ApiTransactionResult{
+					Type:     "normal",
+					Sender:   hex.EncodeToString(tx.SndAddr),
+					Receiver: hex.EncodeToString(tx.RcvAddr),
+				}, nil
+			},
+		}
+		metaAPIBlockProc.txStatusComputer = &mock.StatusComputerStub{}
+
+		metaAPIBlockProc.logsFacade = &testscommon.LogsFacadeStub{
+			IncludeLogsInTransactionsCalled: func(_ []*transaction.ApiTransactionResult, _ [][]byte, _ uint32) error {
+				return nil
+			},
+		}
+		metaAPIBlockProc.alteredAccountsProvider = &testscommon.AlteredAccountsProviderStub{
+			ExtractAlteredAccountsFromPoolCalled: func(txPool *outportcore.Pool, options shared.AlteredAccountsOptions) (map[string]*outportcore.AlteredAccount, error) {
```

Comment:
I forgot why we also need to fetch data for accounts involved in txpool transactions (here on the block API) - why don't we get the addresses from transactions in block's miniblocks?

Reply:
Sorry for the confusion. Perhaps rename `txPool` to `outportPool`?

Reply:
I wanted to reuse as many code as possible. renamed